tests/test_tui_publish_workflow_security.py

```
@@ -1,13 +1,450 @@
+    for name, test in sorted(globals().items()):
+        if name.startswith("test_") and callable(test):
+            suite.addTest(unittest.FunctionTestCase(test, description=name))
+    return suite
+
+
 def workflow(name: str) -> str:
     return (ROOT / ".github" / "workflows" / name).read_text()
 
 
+def workflow_triggers(text: str) -> dict[str, object]:
+    document = yaml.load(text, Loader=yaml.BaseLoader)
```

Ruff (0.16.0)

[error] Probable use of unsafe loader `BaseLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)